src/main.rs

```diff
@@ -213,37 +234,36 @@ fn reader_for(path: &Path) -> anyhow::Result<BufReader<File>> {
 fn load_cert_and_key(
 	cert_path: &Path,
 	key_path: &Path
-) -> anyhow::Result<(Vec<Certificate>, PrivateKey)> {
-	let certificates = rustls_pemfile::read_all(&mut reader_for(cert_path)?)
-		.with_context(|| format!("Failed to parse {}", cert_path.display()))?
-		.iter()
-		.filter_map(|cert| match cert {
-			rustls_pemfile::Item::X509Certificate(cert) => {
-				Some(Certificate(cert.to_owned()))
-			},
-			_ => None
+) -> anyhow::Result<Arc<impl ResolvesServerCert>> {
+	let certificates = rustls_pemfile::certs(&mut reader_for(cert_path)?)
+		.filter_map(|cert| {
+			match cert.with_context(|| format!("Failed to parse {}", cert_path.display()))
+			{
+				Ok(s) => Some(s),
+				Err(_) => None
+			}
```

Comment:
Right here, we construct a `Result` with context just to throw it away and make it an option (which btw can be done by calling `.ok()`). I think the best option here would be to issue a `warn` log message with the error.